src/services/tx/tx-sender/sdk.ts

```diff
@@ -29,3 +35,58 @@ export const getUncheckedSafeSDK = (provider: Web3Provider): Promise<Safe> => {
 
   return sdk.connect({ ethAdapter })
 }
+
+type SigningMethods = Parameters<Safe['signTransaction']>[1]
+
+export const _getSupportedSigningMethods = (
+  safeVersion: SafeInfo['version'],
+  wallet: ConnectedWallet,
+): SigningMethods[] => {
+  const SAFE_VERSION_SUPPORTS_ETH_SIGN = '>=1.1.0'
+
+  if (!safeVersion || !semverSatisfies(safeVersion, SAFE_VERSION_SUPPORTS_ETH_SIGN)) {
+    return ['eth_signTypedData_v4']
+  }
+
+  if (shouldUseEthSignMethod(wallet)) {
+    return ['eth_sign']
+  }
+
+  // Hardware wallet may be connected via MetaMask so we try `eth_sign` after EIP-712
+  return ['eth_signTypedData_v4', 'eth_sign']
```

Comment:
An enum for these would be good.

Reply:
As discussed, I'll keep this as a string now that I've reduced the returned methods as the signing method argument of `signTransaction` [is a string union](https://github.com/safe-global/safe-core-sdk/blob/main/packages/safe-core-sdk/src/Safe.ts#L442-L446).

Reply:
Two constants would be good then, to not repeat the string literal twice.